presto-spi/src/main/java/com/facebook/presto/spi/statistics/ExternalPlanStatisticsProvider.java

```diff
@@ -13,17 +13,32 @@
  */
 package com.facebook.presto.spi.statistics;
 
+import com.facebook.presto.common.plan.PlanCanonicalizationStrategy;
 import com.facebook.presto.spi.plan.PlanNode;
 import com.facebook.presto.spi.plan.TableScanNode;
 
+import java.util.Map;
 import java.util.function.Function;
 
 public interface ExternalPlanStatisticsProvider
 {
     String getName();
 
-    PlanStatistics getStats(
-            PlanNode plan,
+    /**
+     * Get historical statistics for given plan nodes.
+     * Depending on `useHashBasedStatistics`, implementation may choose to use plan node hashes supplied,
+     * or do its own statistics prediction by reading the plan.
+     *
+     * TODO: Using PlanNode as map key may be expensive, we can use PlanNodeId as map keys
+     */
+    Map<PlanNode, HistoricalPlanStatistics> getStats(
+            Map<PlanNode, Map<PlanCanonicalizationStrategy, String>> planNodeHashes,
+            boolean useHashBasedStatistics,
             Function<PlanNode, String> planPrinter,
             Function<TableScanNode, TableStatistics> tableStatisticsProvider);
```

Comment:
I know these two are from previous PRs. Could you elaborate more on the use of these two params?